### ISSUE TITLE
Add new poetry extras for rest sigv4

### ISSUE THREAD

**Issue by paulmech** [apache/iceberg-python#1405]
### Feature Request / Improvement

Currently, rest catalog support appears built in to the main `pyiceberg` package

However, rest catalog also supports SIGv4 signing, without an explicit dependency on boto3, which is required to fetch existing credentials and create the SIGv4 Authorization header. 

With AWS Glue now adding support for [Iceberg REST endpoints](https://github.com/apache/iceberg-python/blob/main/pyiceberg/catalog/rest.py) (seemingly unrelated to s3tables), it is possible to access a glue catalog using the rest catalog interface, but it requires boto3 for the SIGv4 signing. E.g. in my `~./.pyiceberg.yaml` I have

```yaml
catalog:
  default:
    uri: https://glue.us-west-2.amazonaws.com/iceberg
    rest.sigv4-enabled: true
    rest.signing-region: us-west-2
    rest.signing-name: glue
```

It works, but I have to add `boto3` package first.

Suggestion is to add a poetry extra `rest-sigv4` that adds boto3 dependency.

**Comment by kevinjqliu** [comment]
thanks for raising this issue! It does look like REST Catalog's SIGv4 feature requires boto3
https://github.com/apache/iceberg-python/blob/bfc0d9a62176803094da0867ee793808f105d352/pyiceberg/catalog/rest.py#L474-L477

Is this something you would like to contribute? I think we need to add the poetry extra [here](https://github.com/apache/iceberg-python/blob/bfc0d9a62176803094da0867ee793808f105d352/pyproject.toml#L873-L874)

**Comment by Fokko** [comment]
Ah, this is a great catch! Boto comes with S3FS, and if you don't use that, you have to install it separately.

**Comment by kevinjqliu** [comment]
Closed by #1415
